model/market.ts

```diff
@@ -357,3 +359,75 @@ export const calculateDepositApy = (
     proceeds: totalDeposit - initialDeposit,
   }
 }
+
+export const calculateBorrowApr = (
+  substitute: Currency,
+  markets: Market[],
+  initialBorrow: bigint,
+  maxAmountExcludingFee: bigint,
+  currentTimestamp: number,
+): {
+  interest: bigint
+  maxInterest: bigint
+  apr: number
+  totalBorrow: bigint
+  available: bigint
+} => {
+  if (
+    markets.some(
+      (market) =>
+        !isAddressEqual(
+          market.quoteToken.address,
+          substitute.address as `0x${string}`,
+        ),
+    )
+  ) {
+    new Error('Substitute token is not supported')
+  }
+
+  const availableCoupons = min(
+    ...markets.map((market) => market.totalAsksInBaseAfterFees()),
+  )
+  const interestAvailableCoupons = min(
+    availableCoupons -
+      markets.reduce(
+        (acc, market) =>
+          acc + market.take(substitute.address, availableCoupons).amountIn,
+        0n,
+      ),
+    0n,
+  )
+
+  const maxInterest = markets.reduce(
+    (acc, market) =>
+      acc + market.take(substitute.address, maxAmountExcludingFee).amountIn,
+    0n,
+  )
+  let interest = 0n
+  const prevInterests = new Set<bigint>()
+  while (!prevInterests.has(interest)) {
+    prevInterests.add(interest)
+    interest = markets.reduce(
+      (acc, market) =>
+        acc +
+        market.take(substitute.address, initialBorrow + interest).amountIn,
+      0n,
+    )
+  }
+
+  const endTimestamp = markets
+    .map((market) => market.endTimestamp)
+    .reduce((acc, val) => (acc < val ? val : acc), 0)
```

Comment:
`Math.max`

Reply:
I've fixed both the calculateDepositApy function: https://github.com/clober-dex/coupon.finance/pull/74/commits/06654449e5899c1c453ec09ad7f71e43d29cbc38